### DISCUSSION TITLE
Tracking a triggered workflow run

### DISCUSSION THREAD

**Discussion by Anwar-Haque** [community/community#46372] [Apps, API and Webhooks]
How can we track the api dispatch workflow to trace the progress of triggered workflow?

No id or information pertaining to the workflow we just triggered?

How do we track the workflow we triggered from outside of the platform?

What if we need an artifact/output generated from the Workflow?

**Comment by chris-c-thomas** [comment]
I think you're going to want to take a look at these sets of docs for some specific ways to interact with the API. You can track the workflow and see what's happening.

Docs for:
[Workflows](https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28)
[Workflow runs](https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28)
[Workflow jobs](https://docs.github.com/en/rest/actions/workflow-jobs?apiVersion=2022-11-28)
[Artifacts to share and store data](https://docs.github.com/en/rest/actions/artifacts?apiVersion=2022-11-28)
[Webhooks](https://docs.github.com/en/developers/webhooks-and-events/webhooks/about-webhooks)
[GitHub event types](https://docs.github.com/en/developers/webhooks-and-events/events/github-event-types)

Do you have an example or specific idea that we could work with to narrow this down and give you a better solution?

**Reply by Anwar-Haque** [reply]
Thank you Chris. I am looking specifically when we trigger an external workflow through the dispatch workflow API .

Calling workflow just triggers the external workflow however, in response we get 204.

It's hard to identify if the external workflow is gracefully completed or failed we are using curl command to trigger and do not want to attach with run_id/tags as it becomes complicates to manage for 1000+ workflows.

Please share if there is a working example workflow to refer too.

**Reply by chris-c-thomas** [reply]
Instead of checking the response code can you check the `state` of the workflow? When it's running it should be `active` and when it's completed I believe it should be `inactive`. I know `disabled_manually` also exists or something along those lines.

**Reply by Anwar-Haque** [reply]
Thank you Chris good idea but we have multiple triggering sources at any point say parallel execution will be 10-15 jobs. trigger the workflow.

Now the question comes the triggering workflow will not know it's triggered job status is successful or failed same for all other source triggers. 

Simply put we want to implement the tracing end to end workflow with its triggered job status updates to source workflow so that it knows that there is not just triggered workflow but also gives status

**Reply by chris-c-thomas** [reply]
Ahh I see. Let me think on this a little bit today. I'll come back soon @Anwar-Haque.

**Reply by Anwar-Haque** [reply]
Surely @chris-c-thomas thanks in advance 🙂

**Comment by cihad** [comment]
We strongly expect the workflow dispatch event to return the `run_id`.

**Comment by mat007** [comment]
Similar to https://github.com/orgs/community/discussions/17389 and https://github.com/orgs/community/discussions/9752
